test/myOwnGraph/GraphTest.java

```diff
@@ -0,0 +1,85 @@
+package myOwnGraph;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThrows;
+
+import org.junit.Test;
+
+public class GraphTest {
+
+	
+	@Test
+	public void insertEdgeTest() {
+		
+		Edge edge = new Edge();
+		
+		Vertex vertexU = new Vertex("u".charAt(0));
+		
+		Vertex vertexV = new Vertex("v".charAt(0));
+		
+		Graph graph = new Graph();
+		
+		assertEquals(edge,graph.insertEdge(edge, vertexU, vertexV));
+		
+	}
+	
+	@Test
+	public void insertEdgeExceptionTest(){
+		
+		Edge edge = new Edge();
+		
+		Vertex vertexU = new Vertex("u".charAt(0));
+		
+		Vertex vertexV = new Vertex("v".charAt(0));
+		
+		Graph graph = new Graph();
+		
+		graph.insertEdge(edge, vertexU, vertexV);
+		
+		assertThrows(IllegalArgumentException.class,() ->graph.insertEdge(edge, vertexU, vertexV));
+		
+		
+	}
+	
+	@Test
+	public void getEdgeTest(){
+		
+		Edge edge = new Edge();
+		
+		Vertex vertexU = new Vertex("u".charAt(0));
+		
+		Vertex vertexV = new Vertex("v".charAt(0));
+		
+		Graph graph = new Graph();
+		
+		graph.insertEdge(edge, vertexU, vertexV);
+		
+		assertEquals(edge, graph.getEdge(vertexU, vertexV));
+		
+		
+	}
+	
+	@Test
+	public void getEdgeNullTest(){
+		
+		Vertex vertexU = new Vertex("u".charAt(0));
+		
+		Vertex vertexV = new Vertex("v".charAt(0));
+		
+		Vertex vertexA = new Vertex("a".charAt(0));
+		
+		Vertex vertexB = new Vertex("b".charAt(0));
+		
+		Edge edge = new Edge();
+		
+		Graph graph = new Graph();
+		
+		graph.insertEdge(edge, vertexA, vertexB);
+			
+		assertNull(graph.getEdge(vertexU, vertexV));
```

Comment:
//we do not need lines 66-76 anymore
//since the edge between vertices u and v was created previously, my suggestion is to use this test
assertNull(graph.getEdge(vertexA, vertexB));

Reply:
I left 70 and 72 in here since VertexA and VertexB aren't used anywhere else. Does that work ?

Reply:
can you provide more details on this one? I am not sure...

Reply:
Since those two vertices are only used in one test instantiate them here rather than in the setup method.